### ISSUE TITLE
Nested heredoc syntax highlighting is broken

### ISSUE THREAD

**Issue by softwarecreations** [atom/language-shellscript#165]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description
Nested heredoc syntax highlighting is broken.

### Steps to Reproduce
Simply paste the code and see.

1. Create `script1.bash` as follows
```
#!/bin/bash
echo "Script 1 starting"
echo These files exist: script*.bash
cat >script2.bash << SCRIPT2
#!/bin/bash
echo "Script 2 starting"

cat >script3.bash << SCRIPT3
  echo "Script 3 starting"
  foo="So broken"
  echo "This is $foo" #this is highlighted (bad)
  totallyBroken() {
    echo "Yep"
  }
  foo1="$(totallyBroken)"  #not highlighted (good)
  foo11="$(totallyBroken)" #highlighted (bad)
   foo1="$(totallyBroken)" #highlighted (bad)... it seems the horizontal position of totallyBroken is playing a role
  echo "Script 3 ending"
SCRIPT3
echo These files exist: script*.bash
bash script3.bash
echo "Script 2 ending"
SCRIPT2
echo These files exist: script*.bash
bash script2.bash
echo "Script 1 ending"
```
2. You can run it if you want, to confirm that it works: `bash script1.bash`

**Expected behavior:**
**Everything** between SCRIPT2 and SCRIPT2 should be green or whatever means embedded heredoc.

**Actual behavior:**
1. The number 2 at the start of SCRIPT2 is highlighted, which makes absolutely no sense.
2. The number 3 at the end of SCRIPT3 is highlighted, which makes even less sense.
3. echo, bash and echo inside SCRIPT2 are highlighted.

Basically language-shellscript doesn't work properly. It's supposed to parse the heredoc properly, using SCRIPT2 as the open and SCRIPT2 as the close, but instead it assumes that Heredoc symbols can only be alphabetic characters.

If you swap SCRIPT2 with SCRIPTTWO or SCRIPTtwo then it stops fumbling the nested Heredocs but continues to fumble the variable and function call inside the heredoc.

**Reproduces how often:**
Every time 

### Versions
Atom    : 1.54.0
Electron: 6.1.12
Chrome  : 76.0.3809.146
Node    : 12.4.0
apm  2.5.2
npm  6.14.8
node 12.4.0 x64
atom 1.54.0
python 2.7.16
git 2.29.2
Linux Debian Bullseye

### Additional Information
N/A
